### DISCUSSION TITLE
(event)  On Rewarded Cancelled

### DISCUSSION THREAD

**Discussion by EMI-INDO** [admob-plus/admob-plus#434] [Q&A]
On Rewarded Cancelled
Triggers when a rewarded advert has been cancelled.

`document.addEventListener('admob.rewarded.reward', async (event) => { 

                if (event)
                {
                    alert("On Rewarded Complete")
                } else   {
                   
                     alert("On Rewarded Cancelled")
                }
                
              
            });`

### Can you make a few examples?

Help is much appreciated

Thank you friend

**Comment by ratson** [comment]
@EMI-INDO You may listen to the [`dismiss`](https://admob-plus.github.io/docs/cordova/ads/rewarded#dismiss) event when there is no reward.

**Comment by kreso22** [comment]
I'm sorry @ratson , I'm running into the same problem.
I feel like I am doing something very obviously wrong, but after hours of testing can't figure it out :P

Te only events being reported for me are 'load' and 'loadfail'.
'show', 'showfail', 'dismiss'... are never triggered.

```
var admobRewardedAd = new admob.RewardedAd({adUnitId: ###});

admobRewardedAd.on('load', (...e) => {
    // this is the only event that is reported (and loadfail)
});

admobRewardedAd.on('dismiss', (...e) => {
    // does NOT work
});

admobRewardedAd.on('admob.ad.dismiss', (...e) => {
    // does NOT work
});

document.addEventListener('admob.ad.dismiss', (...evt) => {
    // does NOT work
});
```

I switch between programming languages often, I am not even sure what is the correct syntax any more.
Could you create an explicit example? I would really appreciate it.

**Reply by kreso22** [reply]
I'll respond to my own concern. 
The problem I was running into was because I was running `admobRewardedAd.load()` immediately after showing it.
This in effect disabled all of the events.

This is example of code that didn't work:

```
admobInterstitialAd.show().then( () => {
    continueGame();
    admobInterstitialAd.load(); // after I removed this line, events got fired properly
});
```

**Reply by itple-sw** [reply]
Can I see all your code? I can't use rewarded ads in my app.

**Reply by kreso22** [reply]
> Can I see all your code? I can't use rewarded ads in my app.

No you can't see my code but perhaps I can help you with your implementation :)
Show me your code and maybe I can help.
